ir/function.cpp

```diff
@@ -289,4 +289,66 @@ void CFG::printDot(ostream &os) const {
   os << "}\n";
 }
 
+void DomTree::buildDominators() {
+  function<void(const BasicBlock*)> ensureExists;
+  ensureExists = [&](auto bb) {
+    if (dominators.find(bb) == dominators.end()) {
+      std::vector<const BasicBlock*> v{bb};
+      dominators.emplace(bb, move(v));
```

Comment:
this whole function is equivalent to: `dominators.try_emplace(bb)`
Though it doesn't seem really necessary as you can just have a variable like `auto &src_dom = dominators[&src]`